src/comm/util.js

```diff
@@ -14,10 +14,146 @@
 
 'use strict';
 const path = require('path');
+const fs = require('fs');
+const moment  = require('moment');
+const winston = require('winston');
 
 // comm --> src --> root
 const rootDir = path.join('..', '..');
+const LOGGING_LEVELS =  ['debug','info','warn','error'];
+const maxSize = 5242880; //5M
 
+/**
+ * Save the logging configurations
+ * @param {object} logger  cutomed winston logger
+ */
+function saveLogger(logger) {
+    if (global.caliper) {
+        global.caliper.logger = logger;
+    } else {
+        global.caliper = {
+            logger: logger
+        };
+    }
+}
+
+/**
+ * When starting to run tests, the old log files with the default names and
+ * the current log cofigurations' name should be deleted.
+ * @param {String} fileName  in default creating mode, this name is
+ *                corresponding to log level without '.log'; otherwise, this
+ *                is a full name with '.log'
+ */
+function deleteLoggerFiles(fileName) {
+    let curLogName;
+    let baseName;
+    if(fileName.indexOf('.log')>=0 && fileName.indexOf('.log')===fileName.length -4){
+        curLogName = fileName;
+        baseName = path.basename(fileName,'.log');
+    }
+    else{
+        curLogName = fileName +'.log';
+        baseName = fileName;
+    }
+    let logNumber = 0;
+    //remove the last file;
+    if(fs.existsSync(curLogName)){
+        fs.unlinkSync(curLogName);
+    }
+    logNumber++;
+    curLogName = baseName + logNumber.toString() +'.log';
+    while(fs.existsSync(curLogName)){
+        fs.unlinkSync(curLogName);
+        logNumber++;
+        curLogName = baseName + logNumber.toString() +'.log';
+    }
+}
+
+/**
+ * define a log format
+ * @param {object} options winston's options for log definition
+ * @returns {string} a formatted string
+*/
+function logFormat(options){
+    return  options.timestamp() + ' - ' +
+    options.level.toUpperCase() +
+    ' ' +(options.message ? options.message : '') +
+    (options.meta && Object.keys(options.meta).length ? '\n\t' + JSON.stringify(options.meta) : ' ') +
+    '\r\n';
+}
+
+/**
+ * If there is no  logging configurations, create a new logger,
+ * it will generater 3 files named 'info.log', 'error.log','warn.log'.
+ * The debug information is not saved to a file, it will just show to console.
+ * @return {object}   the new winston logger
+ */
+function newDefaultLogger() {
+    let transports =[new (winston.transports.Console)({
+        colorize: true})];
+    ['debug', 'info', 'warn', 'error'].forEach((level)=>{
+        try {
+            //remove the last file;
+            deleteLoggerFiles(level);
+            if(level==='debug'){
```

Comment:
I think the default should be something like the following:
* info (which is essentially progress report) and above (warning, error) should be printed to the console
* debug should go to a debug file